runtime/interpreter/interpreter_expression.go

```diff
@@ -523,6 +497,83 @@ func (interpreter *Interpreter) testEqual(left, right Value, expression *ast.Bin
 	)
 }
 
+func (interpreter *Interpreter) testComparison(left, right Value, expression *ast.BinaryExpression) BoolValue {
+	left = interpreter.Unbox(
+		LocationRange{
+			Location:    interpreter.Location,
+			HasPosition: expression.Left,
+		},
+		left,
+	)
+
+	right = interpreter.Unbox(
+		LocationRange{
+			Location:    interpreter.Location,
+			HasPosition: expression.Right,
+		},
+		right,
+	)
+
+	leftComparable, ok := left.(ComparableValue)
+	if !ok {
+		return FalseValue
```

Comment:
This was previously a panic with `InvalidOperandsError`, see the local `error` function in the calling function: https://github.com/onflow/cadence/pull/2430/files#diff-484f7e68f1a1b9f96531632abacc84c131416a281cb29a502ac3b98c041ddb78R296-R303

Reply:
Thanks for the pointer. Updated the logic.